crates/bevy_sprite/src/render/sprite_sheet.vert

```diff
@@ -45,7 +46,22 @@ void main() {
         vec2(sprite_rect.end.x, sprite_rect.begin.y), 
         sprite_rect.end
     );
-    v_Uv = (atlas_positions[gl_VertexIndex]) / AtlasSize;
-    v_Color = TextureAtlasSprite_color;
+
+    vec2 uv = (atlas_positions[gl_VertexIndex]) / AtlasSize;
+
+    // Flip the sprite if necessary
+    uint x_flip_bit = 1;
+    uint y_flip_bit = 2;
+
+    if ((flip & x_flip_bit) == x_flip_bit) {
+        uv = vec2(1.0 - uv.x , uv.y);
+    }
+    if ((flip & y_flip_bit) == y_flip_bit) {
+        uv = vec2(uv.x, 1.0 - uv.y);
+    }
```

Comment:
Found out that this logic isn't right. It just looked like it worked with the one-line spritesheet we have in the bevy assets folder. I'll work on fixing it.

Reply:
Fixed it. :smiley: